File: OCH_Tao/ABC/B/175.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const L = input[1].split(" ").map(x => parseInt(x, 10));
let cnt = 0;
for (let i = 0; i < N; i++) {
  for (let j = i + 1; j < N; j++) {
    for (let k = j + 1; k < N; k++) {
      if (L[i] !== L[j] && L[j] !== L[k] && L[k] !== L[i]) {
        if (Math.max(L[i], L[j], L[k]) < L[i] + L[j] + L[k] - Math.max(L[i], L[j], L[k])) {
          cnt++;
        }
      }
    }
  }
}
console.log(cnt);

File: OCH_Tao/ABC/B/175.py
N = int(input())
L = list(map(int,input().split()))
cnt = 0
for i in range(N):
  for j in range(i+1,N):
    for k in range(j+1,N):
      if L[i]!=L[j] and L[j]!=L[k] and L[k]!=L[i] and (X:=max(L[i],L[j],L[k]))<L[i]+L[j]+L[k]-X:
        cnt+=1
print(cnt)